colossus/src/main/scala/colossus/controller/Controller.scala

```diff
@@ -42,6 +41,8 @@ trait MasterController[Input, Output] extends ConnectionHandler {
   protected def codec: Codec[Output, Input]
   protected def controllerConfig: ControllerConfig
 
+  implicit def namespace : metrics.MetricNamespace
```

Comment:
This one is abstract, but which namespace it uses depends on whether it's a server or client.